bundles/org.openhab.binding.tapocontrol/src/main/java/org/openhab/binding/tapocontrol/internal/devices/wifi/TapoBaseDeviceHandler.java

```diff
@@ -132,29 +137,39 @@ private void activateDevice() {
     }
 
     /**
-     * CHECK SETTINGS
-     *
-     * @return TapoErrorHandler with configuration-errors
+     * Check if bridge is set
      */
-    protected TapoErrorHandler checkSettings() {
-        TapoErrorHandler configErr = new TapoErrorHandler();
-
+    protected boolean checkBridge() {
         /* check bridge */
         if (!(bridge instanceof TapoBridgeHandler)) {
-            configErr.raiseError(ERR_CONFIG_NO_BRIDGE);
-            return configErr;
-        }
-        /* check ip-address */
-        if (!config.ipAddress.matches(IPV4_REGEX)) {
-            configErr.raiseError(ERR_CONFIG_IP);
-            return configErr;
+            setError(new TapoErrorHandler(ERR_CONFIG_NO_BRIDGE));
+            return false;
         }
         /* check credentials */
         if (!bridge.getCredentials().areSet()) {
-            configErr.raiseError(ERR_CONFIG_CREDENTIALS);
-            return configErr;
+            setError(new TapoErrorHandler(ERR_CONFIG_CREDENTIALS));
+            return false;
         }
-        return configErr;
+        return true;
+    }
+
+    /**
+     * Check Device Configuration
+     */
+    protected boolean checkConfig() {
```

Comment:
I personally prefer to make the configuration class responsible to check if it is valid or not. 
Maybe something to think about, it would make this class also more readable.

Reply:
Good idea. I will change it